### PR TITLE
aws-signing-helper: init at 1.3.0, 1.3.0 -> 1.4.0

### PR DESCRIPTION
**Pull request by pandanz**
Adds a package for [rolesanywhere-credential-helper](https://github.com/aws/rolesanywhere-credential-helper). This is a go application which is used to obtain temporary security credentials in IAM for workloads such as servers, containers, and applications that run outside of AWS.

[AWS Docs](https://docs.aws.amazon.com/rolesanywhere/latest/userguide/introduction.html)  

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pandanz** [comment]
@steeleduncan, Can you please review? I accidentally closed the previous PR.

File: maintainers/maintainer-list.nix
Comment (steeleduncan) [inline maintainers/maintainer-list.nix]: This change should be squashed into the maintainer commit to keep it clean

File: pkgs/by-name/aw/aws-signing-helper/package.nix
Comment (steeleduncan) [inline pkgs/by-name/aw/aws-signing-helper/package.nix:14]: `hash = "sha256-1488l381449b4z3y5zyj121drax1v0ib0xhsmdy1prsmfck1k3xx"`

**Review by steeleduncan — CHANGES_REQUESTED** [review]

**Comment by steeleduncan** [comment]
Thanks for making those changes

Additionally, the two commits should be named
- `maintainers: add pandanz`
- `aws-signing-helper: init at 1.3.0`

I don't know the package, so it is up to you, but would `rolesanywere-credential-helper` not be the obvious name for this package? given that is the name of the github repo it is drawn from?

File: maintainers/maintainer-list.nix
Comment (pandanz) [reply maintainers/maintainer-list.nix]: Fixed

File: pkgs/by-name/aw/aws-signing-helper/package.nix
Comment (pandanz) [reply pkgs/by-name/aw/aws-signing-helper/package.nix]: Fixed

File: pkgs/by-name/aw/aws-signing-helper/package.nix
Comment (steeleduncan) [inline pkgs/by-name/aw/aws-signing-helper/package.nix:14]: Apologies, now the format has changed you would need to recalculate this

I just checked and the value should be `sha256-vY8ZJnNV5xt8qxp2sCLYoavcggjS/+LHJysREtCgCJE=`

**Review by steeleduncan — CHANGES_REQUESTED** [review]
Even after fixing the hash and rebuilding, I noticed this fails at `checkPhase`

You can turn this off with `doChecks=false;`, but you'd need to provide a comment with a valid reason why they dont work

The error is `fork/exec /bin/bash: no such file or directory`, which implies that something expects FHS, when it is of course not present within a nix build. Potentially you could use `buildFHSEnv` to make the tests run properly?

**Comment by steeleduncan** [comment]
> what do I need to do to get this over the line?

Once the various checks on the PR are showing green (either because you disable them internally, or fix them), you will need someone with the commit bit to merge it

My suggestion would be to check the recent commit history of `nixpkgs` and find someone who has merged a broadly similar PR recently (e.g. golang based), and request a review from them via GitHub

File: pkgs/by-name/aw/aws-signing-helper/package.nix
Comment (pandanz) [reply pkgs/by-name/aw/aws-signing-helper/package.nix]: Thanks, fixed

File: pkgs/by-name/aw/aws-signing-helper/package.nix
Comment (steeleduncan) [inline pkgs/by-name/aw/aws-signing-helper/package.nix:20]: `runHook preCheck`

File: pkgs/by-name/aw/aws-signing-helper/package.nix
Comment (steeleduncan) [inline pkgs/by-name/aw/aws-signing-helper/package.nix:22]: `runHook postCheck`

**Review by steeleduncan — APPROVED** [review]
This needs hooks now you are overriding the check phase, but otherwise looks good to me assuming ofborg is ok with it!

**Comment by cole-h** [comment]
I just pushed a couple of changes to this branch:

1. Renamed the commit adding you as a maintainer to match the expected format (`maintainers: add pandanz`)
2. Added a commit updating to the most recent version, 1.4.0

Once CI is happy, I'll merge :)

**Review by cole-h — APPROVED** [review]

**Pull request merged**
